driftbase/match_placements.py

```diff
@@ -0,0 +1,539 @@
+                "placement_id": placement_id,
+                "player_id": player_id,
+                "match_provider": match_provider,
+                "lobby_id": lobby_id,
+                "status": "pending",
+                "create_date": datetime.datetime.utcnow().isoformat(),
+            }
+
+            with _JsonLock(_get_match_placement_key(placement_id)) as match_placement_lock:
+                match_placement_lock.value = match_placement
+
+            with _GenericLock(_get_player_match_placement_key(player_id)) as player_match_placement_lock:
+                player_match_placement_lock.value = placement_id
+
+            lobby["status"] = "starting"
+            lobby["placement_date"] = datetime.datetime.utcnow().isoformat()
+
+            lobby_lock.lobby = lobby
+
+            log.info(f"GameLift game session placement issued for lobby '{lobby_id}' by host player '{player_id}'")
+
+            # Notify members
+            receiving_player_ids = _get_lobby_member_player_ids(lobby, [player_id])
+            _post_lobby_event_to_members(receiving_player_ids, "LobbyMatchStarting", {"lobby_id": lobby_id, "status": lobby["status"]})
+
+            return match_placement
+
+def stop_player_match_placement(player_id: int, expected_match_placement_id: str):
+    with _GenericLock(_get_player_match_placement_key(player_id)) as player_match_placement_lock:
+        placement_id = player_match_placement_lock.value
+
+        if expected_match_placement_id != placement_id:
+            log.warning(f"Player '{player_id}' attempted to stop match placement '{expected_match_placement_id}', but the player didn't issue the match placement")
+            raise UnauthorizedException(f"You don't have permission to access match placement {expected_match_placement_id}")
+
+        with _JsonLock(_get_match_placement_key(placement_id)) as match_placement_lock:
+            placement = match_placement_lock.value
+
+            if placement:
+                lobby_id = placement.get("lobby_id", None)
+                if not lobby_id:
+                    raise RuntimeError(f"Match placement '{placement_id}' doesn't contain a 'lobby_id' field. Only lobby match placements are supported at this time")
+
+                match_provider = placement["match_provider"]
+                if match_provider != "gamelift":
+                    raise RuntimeError(f"Invalid match provider configured, '{match_provider}'. Only the GameLift match provider is supported at this time")
+
+                placement_status = placement["status"]
+                if placement_status != "pending":
+                    log.warning(f"Player '{player_id}' attempted to stop match placement '{expected_match_placement_id}', but the placement is in status '{placement_status}'")
+                    raise InvalidRequestException(f"Cannot stop a match placement in status {placement_status}")
+
+                response = flexmatch.stop_game_session_placement(placement_id)
```

Comment:
This can fail, no?  try/except needed?

Reply:
Yes, the flexmatch.stop_game_session_placement has guards for the GameLift exceptions and re-raises them as our homebrew ones.